src/main/java/org/broadinstitute/hellbender/utils/pairhmm/SampleSitesForDRAGstrModel.java

```diff
@@ -0,0 +1,437 @@
+        return true;
+    }
+
+    private File tempDir;
+
+    private long[][] totalCounts;
+
+    private long[][] emittedCounts;
+
+    public void onStartup() {
+        super.onStartup();
+        try {
+            tempDir = File.createTempFile("gatk-sample-dragstr-sites", ".tmp");
+        } catch (final IOException ex) {
+            throw new GATKException("could not create temporary disk space", ex);
+        }
+        if (!tempDir.delete()) {
+            throw new GATKException("could not create temporary disk space: could not delete tempfile");
+        } else if (!tempDir.mkdir()) {
+            throw new GATKException("could not create temporary disk space: could not create tempdir");
+        }
+        totalCounts = new long[maxPeriod + 1][maxRepeat + 1];
+        emittedCounts = new long[maxPeriod + 1][maxRepeat + 1];
+    }
+
+    public void onShutdown() {
+        Utils.deleteFileTree(tempDir);
+        super.onShutdown();
+    }
+
+    @Override
+    public void traverse() {
+        final SAMSequenceDictionary dictionary = Utils.nonNull(getReferenceDictionary());
+        final File tempDir = this.tempDir;
+        try (final AutoCloseableList<BinaryTableWriter<DragstrLocus>> allSitesWriter
+                     = AutoCloseableList.of(maxPeriod, i -> createDragstrLocusWriter(tempDir, "period_" + (i + 1) + ".bin"))) {
+            if (!intervalArgumentCollection.intervalsSpecified()) {
+                for (final SAMSequenceRecord sequence : dictionary.getSequences()) {
+                    traverse(sequence.getSequenceIndex(), sequence, 1, sequence.getSequenceLength(), allSitesWriter, decimationTable);
+                }
+            } else {
+                for (final SimpleInterval interval : intervalArgumentCollection.getIntervals(dictionary)) {
+                    final SAMSequenceRecord sequence = dictionary.getSequence(interval.getContig());
+                    traverse(sequence.getSequenceIndex(), sequence, interval.getStart(), interval.getEnd(), allSitesWriter, decimationTable);
+                }
+            }
+        } catch (final IOException ex) {
+            throw new UserException.CouldNotCreateOutputFile(tempDir, ex);
+        }
+        progressMeter.stop();
+        logger.info("Finishing reference sampling. Proceeding to splitting each period case by repeat count");
+        for (int i = 1; i <= maxPeriod; i++) {
+            progressMeter = new ProgressMeter(secondsBetweenProgressUpdates);
+            progressMeter.setRecordLabel("Splitting cases with period " + i + " by repeat count");
+            progressMeter.start();
+            splitPeriodLociByRepeat(i);
+            progressMeter.stop();
+            logger.info("Done with period " + i);
+        }
+        logger.info("Composing output zip");
+        composeOutputZip();
+    }
+
+    private void splitPeriodLociByRepeat(final int period) {
+        final File lociFile = new File(tempDir, "period_" + period + ".bin");
```

Comment:
How big is this list of loci internally? Is it small enough to fit into memory? Its just a list of sites correct?